src/snowflake/snowpark/dataframe.py

```diff
@@ -1011,16 +1011,27 @@ def select(
                     )
                 table_func = e
                 func_expr = _create_table_function_expression(func=table_func)
-                join_plan = self._session._analyzer.resolve(
-                    TableFunctionJoin(self._plan, func_expr)
-                )
 
                 if isinstance(e, _ExplodeFunctionCall):
-                    new_cols = _get_cols_after_explode_join(e, self._plan)
+                    new_cols, alias_cols = _get_cols_after_explode_join(e, self._plan)
                 else:
-                    _, new_cols = _get_cols_after_join_table(
-                        func_expr, self._plan, join_plan
+                    # this join plan is created here to extract output columns after the join. If a better way
+                    # to extract this information is found, please update this function.
+                    temp_join_plan = self._session._analyzer.resolve(
+                        TableFunctionJoin(self._plan, func_expr)
+                    )
+                    _, new_cols, alias_cols = _get_cols_after_join_table(
+                        func_expr, self._plan, temp_join_plan
                     )
```

Comment:
We have temp join here to generate a join sql query and run schema query to get the columns generated after the join. We use the result to do apply aliases into the new generated sql query using `right_cols = new_col_names`. The other way we can do this is to run a describe query on table function but that is practically not very different from the earlier approach. 

If anyone in the team has a better approach to do this, let me know. I'm happy to update this section.

Reply:
I'm not familiar with plan generation, so the only idea I have is to extract the "output column generation" part of the `TableFunctionJoin ` plan generation as a shared function. Note that this might not be easy at all and feel free to ignore if it is not straightforward.